azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1776,12 +1776,49 @@ private void processFlow(ExecutionReference reference,
       }
     }
 
+    /* Helper method to fetch  overriding Executor, if a valid user has specifed otherwise return null */
```

Comment:
update the javadoc to be more understandable?

Also my understanding on the logic is that if anything happens, it will return a null.
Is it possible that you want to return with different types of exceptions (or different exception messages) so that we know what went wrong in this method? (db connection error vs not being able to find specified user specified executor)